packages/synapse-react-client/src/synapse-queries/doi/useDOI.ts

```diff
@@ -89,19 +84,14 @@ export function useCreateOrUpdateDOI(
       }
       const requestDoi = args[1].doi
       if (requestDoi) {
+        const matchingRequestObject: GetRepoV1DoiAssociationRequest = {
+          id: requestDoi.objectId!,
+          type: requestDoi.objectType as DoiObjectType,
+          portalId: requestDoi.portalId,
+          version: requestDoi.objectVersion,
+        }
         queryClient.invalidateQueries({
-          queryKey: keyFactory.getDOIAssociationQueryKey(
-            requestDoi.objectType!,
-            requestDoi.objectId!,
-            requestDoi.objectVersion,
-          ),
-        })
-        queryClient.invalidateQueries({
-          queryKey: keyFactory.getDOIQueryKey(
-            requestDoi.objectType!,
-            requestDoi.objectId!,
-            requestDoi.objectVersion,
-          ),
+          queryKey: keyFactory.getDOIQueryKey(matchingRequestObject),
         })
```

Comment:
Now invalidate the key returned by `getDOIQueryKey`, which matches both of these